docs/topics/Opcodes_and_Status_Codes.md

```diff
@@ -1,100 +1,114 @@
 # Opcodes and Status Codes
 
+## API
+
+In v8 and forward, API form body errors contain both a code and human-readable error message.
+
+###### API Error Codes
+
+| Code                 | Description                                                |
+|----------------------|------------------------------------------------------------|
+| BASE_TYPE_REQUIRED   | "This field is required"                                   |
+| BASE_TYPE_CHOICES    | "Value must be one of {choices}."                          |
+| BASE_TYPE_MIN_LENGTH | "Must be {min_length} or more in length."                  |
+| BASE_TYPE_MAX_LENGTH | "Must be {max_length} or fewer in length."                 |
+| BASE_TYPE_BAD_LENGTH | "Must be between {min_length} and {max_length} in length." |
```

Comment:
this is not a complete collection of error codes, are we posting these just as examples?

Reply:
Oh I went into the constants file and though that was the lot of them? I'll look more

Reply:
JK there's a million not doing that